doc/source/ray-core/patterns/nested-ray-get.rst

```diff
@@ -0,0 +1,25 @@
+.. _nested-ray-get:
+
+Anti-pattern: Calling ray.get on task arguments harms performance
+=================================================================
+
+
+**TLDR:** If possible, pass ``ObjectRefs`` as direct task arguments, instead of passing a list as the task argument and then calling :func:`ray.get() <ray.get>` inside the task.
+
+When a task calls ``ray.get()``, it must block until the value of the ``ObjectRef`` is ready.
+If all cores are already occupied, this can lead to a deadlock, as the task that produce the ``ObjectRef``'s value may need the caller task's resources in order to run.
+To handle this issue, if the caller task would block in ``ray.get()``, Ray temporarily releases the caller's CPU resources to allow the pending task to run.
+This can harm performance and stability because the caller will continue to use a process and memory to hold its stack while other tasks run.
```

Comment:
```suggestion
This behavior can harm performance and stability because the caller continues to use a process and memory to hold its stack while other tasks run.
```